packages/manager/modules/pci/src/projects/project/public-gateways/add/add.routing.js

```diff
@@ -0,0 +1,31 @@
+import { PRODUCT_NAME } from './add.constants';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('pci.projects.project.public-gateways.add', {
+    url: '/new',
+    component: 'pciProjectPublicGatewaysAdd',
+    resolve: {
+      breadcrumb: /* @ngInject */ ($translate) =>
+        $translate.instant('pci_projects_project_public_gateways_add_title'),
+      regions: /* @ngInject */ (
+        projectId,
+        PciPublicGatewaysService,
+        coreConfig,
+      ) =>
+        PciPublicGatewaysService.getRegions(projectId, {
+          ovhSubsidiary: coreConfig.getUser().ovhSubsidiary,
+          product: PRODUCT_NAME.toLowerCase(),
+        }).then((data) => {
+          const filteredProduct = data.products.find((product) => {
+            return product.name === PRODUCT_NAME.toLowerCase()
+              ? product.regions
+              : [];
+          });
+          return filteredProduct.regions.map((region) => {
```

Comment:
**issue(blocking):** If the product is not found, an empty array `[]` is returned . Accessing `map` on `regions` will throw error.